### PR TITLE
feat: add local project quality script

### PR DESCRIPTION
**Pull request by janderssonse**
This PR adds a bash script that runs the same checks that are run in the CI (linting, commits, license).
This makes it possible for a contributor to locally verify the contribution before opening a PR, and so,avoid any CI check surprises later on.

It adds a brief DEVELOPMENT file to explain how to run it.
It also fixed a few minor things found while running the check.

## Checklist

- [x] My contributions and commit messages follows the style guidelines of this project
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] The Pull Request has an informative and human-readable title
- [x] Changes are limited to a single goal (avoid scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] I confirm that I have read any Contribution guidelines (CONTRIBUTING)
- [x] I confirm that I wrote and/or have the right to submit the contents of my PR, by agreeing to the _Developer Certificate of Origin_, by adding a 'sign-off' to my commits

### CODE REVIEW

**Comment by janderssonse** [comment]
Note: dont merge yet, we should finish of #6 first.

**Comment by janderssonse** [comment]
Note: the pr #6  is visible in this pr, until #6 is merged. Just a sidenote, if it at first glance seems like many files diffing to main, most of them are #6 .

**Review by danneleaf — APPROVED** [review]
Went through these changes and it looks good to me!

**Pull request merged**
